### ISSUE TITLE
A mutable jar can't be accessible in OpenShift for remote development

### ISSUE THREAD

**Issue by danieloh30** [quarkusio/quarkus#13629]
**Describe the bug**
(Describe the problem clearly and concisely.)
A mutable jar on OpenShift can't be running with the following error:

Error: Unable to access jarfile /deployments/target/quarkus-run.jar

**Expected behavior**
(Describe the expected behavior clearly and concisely.)
The Quarkus pod(mutable jar) runs well without

**Actual behavior**
(Describe the actual behavior clearly and concisely.)
Pod is CrashLoopBackOff with an Error: Unable to access jarfile /deployments/target/quarkus-run.jar

**To Reproduce**

Link to a small reproducer (preferably a Maven project if the issue is not Gradle-specific).

Or attach an archive containing the reproducer to the issue.

1. Use a getting-started project like helloword
2. Add an quarkus-openshift extension
3. Add the below configurations

**Configuration**
```properties
# Add your application.properties here, if applicable.
quarkus.package.type=mutable-jar
quarkus.container-image.build=true
quarkus.kubernetes-client.trust-certs=true
quarkus.kubernetes.deployment-target=openshift
quarkus.kubernetes.deploy=true
quarkus.openshift.expose=true
```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)
![image](https://user-images.githubusercontent.com/14066977/100880077-50bab580-347a-11eb-9a06-392d2038082c.png)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: OpenJDK 11
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.10.2.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): apache-maven-3.6.3
 - OpenShift: 4.6.3

**Additional context**
(Add any other context about the problem here.)

**Comment by danieloh30** [comment]
@stuartwdouglas

**Comment by geoand** [comment]
Thanks for reporting @danieloh30 .

@iocanel FYI as well.

**Comment by maxandersen** [comment]
just to clarify - your title says "remote development" but I assume you just mean deployment of quarkus app fails in openshift and is not relate to what we refer to as remote devmode, right?

**Comment by danieloh30** [comment]
It's a remote dev mode because a mutable jar needs to be deployed to the OpenShift as running a remote dev mode. However, a mutable jar can't be accessed like the above screenshot when I deployed it via Quarkus OpenShift extension. Hope it helps you clarify the issue. Thanks.

**Comment by geoand** [comment]
@iocanel seems like something up your alley :)

**Comment by danieloh30** [comment]
I just tried to retest this in 1.10.5.Final then it still doesn't work since remote-dev keep trying to trigger the entire S2I processor rather than sending changed classes when I change the code. Also, the environment variable(QUARKUS_LAUNCH_DEVMODE=true) which I set in DC manually is removed during the new S2I processor based on Openshift extension. This occurs connection failure like:

2021-01-04 15:46:29,661 ERROR [io.qua.ver.htt.dep.dev.HttpRemoteDevClient] (Remote dev client thread) Remote dev request failed: java.io.IOException: Server did not start a remote dev session
        at io.quarkus.vertx.http.deployment.devmode.HttpRemoteDevClient$Session.doConnect(HttpRemoteDevClient.java:128)
        at io.quarkus.vertx.http.deployment.devmode.HttpRemoteDevClient$Session.run(HttpRemoteDevClient.java:169)
        at java.base/java.lang.Thread.run(Thread.java:832)

2021-01-04 15:46:29,662 ERROR [io.qua.ver.htt.dep.dev.HttpRemoteDevClient] (Remote dev client thread) Connection failed after 10 retries, exiting

Can we set "QUARKUS_LAUNCH_DEVMODE=true" in application.properties for openshift extension? It takes more time to send a changed code to the remote server(OCP) than I deployed the same quarkus app using a binary build/deployment.
